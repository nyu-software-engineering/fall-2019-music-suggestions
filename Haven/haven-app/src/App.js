import React from 'react';
import { BrowserRouter as Router, Route} from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";

//import logo from './logo.svg';
import './App.css';
import Navbar from "./components/navbar.component";
import SignedOutNavbar from "./components/signed-out-navbar.component";
import Browse from "./components/browse-artists.component";

//import WelcomePage from "./components/welcomepage.component";
import SignIn from "./components/sign-in-page.component";
import CreateUser from "./components/create-user.component";
import UserProfile from './components/user-profile.component.js';
import Gallery from "./components/gallery.component.js";
import MyProfiles from "./components/myprofiles.component.js";
import viewProfile from "./components/view-profile.component.js";

//import FindUser from "./components/find-user.component";
//import UpdateUser from "./components/update-user.component";
//import DeleteUser from "./components/delete-user.component";

function App() {

  return (

    <Router>
      <Navbar/>
      <br/>
      <Route path="/browse" exact component = {Browse} />
      <Route path="/gallery" component = {Gallery}/>
      <Route path="/createProfile" component={UserProfile}/>
      <Route path="/myProfiles" component={MyProfiles}/>
      <Route path="/viewProfile" component={viewProfile}/>
      <Route path="/signin" component={SignIn}/>
      <Route path="/createUser" component={CreateUser}/>
    </Router>
  );



}

export default App;
