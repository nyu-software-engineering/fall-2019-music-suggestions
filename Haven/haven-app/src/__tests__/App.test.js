// run jest --updateSnapshot to update snapshots as components are updated
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from '../App';
import Navbar from '../App';
/*
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
*/

describe('First React component test with Enzyme', () => {
   it('renders without crashing', () => {
      shallow(<App />);
    });
});

describe('Testing Navbar snapshot', () => {
  it("renders correctly", () => {
    const navbar = shallow(<Navbar />);
    expect(navbar).toMatchSnapshot();
  });
});


//practice test
describe('Examining the syntax of Jest tests', () => {

  it('sums numbers', () => {
      expect(1 + 2).toEqual(3);
      expect(2 + 2).toEqual(4);
   });
});
