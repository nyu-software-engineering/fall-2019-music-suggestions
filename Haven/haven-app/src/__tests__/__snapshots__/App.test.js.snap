// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar snapshot renders correctly 1`] = `ShallowWrapper {}`;
